calico_node/startup/startup.go

```diff
@@ -817,6 +822,31 @@ func ensureFelixConfig(c *client.Client, host, key, def string) error {
 	}
 }
 
+// ensureClusterTypeSet ensures that the ClusterType in the Felix global
+// config has the data contained in the passed in clusterType.  This includes
+// merging what was already set in ClusterType with the passed in clusterType.
+func ensureClusterTypeSet(c *client.Client, key, clusterType string) error {
+	global := ""
+	if val, assigned, err := c.Config().GetFelixConfig("ClusterType", global); err != nil {
+		return err
+	} else if !assigned {
+		return c.Config().SetFelixConfig(key, global, clusterType)
+	} else {
+		curCt := strings.Split(val, ",")
+	NEXT:
```

Comment:
FYI we have some similar acode for this stuff here: https://github.com/projectcalico/libcalico-go/blob/master/lib/backend/k8s/k8s.go#L215

Wonder if we can get rid of the NEXT anchor